<launch>

  <!-- THIS WORKS, BUT TESTING OUT OPENCV'S CAMERA DRIVER INSTEAD -->
  <!-- grab camera image feed and publish -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video1" />
    <!-- <param name="video_device" value="/dev/video0" /> -->
    <param name="camera_frame_id" value="usb_cam" />

    <param name="pixel_format" value="yuyv" />
    <!-- <param name="autofocus" value="true" /> -->

    <param name="camera_name" value="external_webcam" />
    <param name="camera_info_url" value="file:///home/njk/Courses/EECS432/Project/ros_ws/src/eecs432_project/calibration/external_webcam.yml" />
    <!-- <param name="camera_info_url" textfile="file://$(find eecs432_project)/calibration/external_webcam.yml" /> -->
    <!-- THE $(find <package>) PATH ISN'T WORKING, WHY? -->
  </node>


  <!-- run image_proc, for webcam -->
  <!-- NOTE: requires camera calibration file to use any interesting topics -->
  <!-- <node name="image_proc" pkg="image_proc" type="image_proc" ns="usb_cam" >
    <param name="camera_info_url" textfile="$(find eecs432_project)/calibration/external_webcam.yml" />
  </node> -->


  <!-- USING OPENCV'S CAMERA DRIVER: -->
  <!-- <node pkg="cv_camera" type="cv_camera_node" name="cv_camera_node">

    <param name="device_id" value="1" />
    <param name="rate" value="30.0" />
    <param name="image_height" type="int" value="480" />
    <param name="image_width" type="int" value="640" />

    <param name="frame_id" value="external_webcam" />
    <param name="camera_info_url" value="file://$(find eecs432_project)/calibration/external_webcam.yml" />
  </node> -->

  <!-- <test test-name="test_cv_camera" pkg="cv_camera" type="test_cv_camera" /> -->

</launch>
